newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+    const eventType = event.getType();
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      gd.asCommentEvent(event).setBackgroundColor(
+        parseInt(backgroundColors.red),
+        parseInt(backgroundColors.green),
+        parseInt(backgroundColors.blue)
+      );
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      gd.asGroupEvent(event).setBackgroundColor(
+        parseInt(backgroundColors.red),
+        parseInt(backgroundColors.green),
+        parseInt(backgroundColors.blue)
+      );
+    }
+  };
+
+  getComment = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      return gd.asCommentEvent(event).getComment();
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      return gd.asGroupEvent(event).getName();
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event type: ' + eventType
+      );
+      return '';
+    }
+  };
+
+  getTextColors = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let textColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      textColors = {
+        red: commentEvent.getTextColorRed(),
+        green: commentEvent.getTextColorGreen(),
+        blue: commentEvent.getTextColorBlue(),
+      };
+      return textColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      var groupEvent = gd.asGroupEvent(event);
+      const r = groupEvent.getBackgroundColorR(),
+        g = groupEvent.getBackgroundColorG(),
+        b = groupEvent.getBackgroundColorB();
+
+      textColors = (r + g + b) / 3 > 200 ? 'black' : 'white'; //Because text color is not supported by Core
+      return textColors;
+    } else {
+      console.error('Dialog was opened for an unsupported event text color.');
+      return '';
+    }
+  };
+
+  getBackgroundColors = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let backgroundColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      backgroundColors = {
+        red: commentEvent.getBackgroundColorRed(),
+        green: commentEvent.getBackgroundColorGreen(),
+        blue: commentEvent.getBackgroundColorBlue(),
+      };
+
+      return backgroundColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      const groupEvent = gd.asGroupEvent(event);
+      backgroundColors = {
+        red: groupEvent.getBackgroundColorR(),
+        green: groupEvent.getBackgroundColorG(),
+        blue: groupEvent.getBackgroundColorB(),
+      };
+
+      return backgroundColors;
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event background color'
+      );
+      return '';
+    }
+  };
```

Comment:
And these 3 methods could be a single `_getInitialStateFromEvent`.
I think this would reduce some boilerplate and the number of if/else to do (once instead of 3 times).

The object you should return is the `State`: `state = this._getInitialStateFromEvent()`

Reply:
I didn't understand return here

Reply:
This return (`return '';`) is incorrect: it's returning a string, while the rest of the function is returning an object. You should then return an object with the same shape (same "type") as the rest of the function.

In other words, surely something like this: `return {r: 0, g: 0, b: 0}; // Return black in case of error.` - though in theory this should never ever be used (in practice, better make the code robust again something passing a wrong event type to this dialog).